src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        if (shoppingCart == null || shoppingCart.getTickets().isEmpty()) {
+            throw new RuntimeException("Shopping cart is empty. Shopping cart: " + shoppingCart);
+        }
+        List<Ticket> tickets = new ArrayList<>(shoppingCart.getTickets());
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setTickets(tickets);
+        order.setOrderDate(LocalDateTime.now());
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        return orderDao.add(order);
```

Comment:
what if adding an order will fail? the cart will become empty. think about order of operations